WordPress/src/main/java/org/wordpress/android/ui/stockmedia/StockMediaPickerActivity.java

```diff
@@ -0,0 +1,590 @@
+        if (show && mSelectionBar.getVisibility() != View.VISIBLE) {
+            AniUtils.animateBottomBar(mSelectionBar, true);
+        } else if (!show && mSelectionBar.getVisibility() == View.VISIBLE) {
+            AniUtils.animateBottomBar(mSelectionBar, false);
+        } else {
+            return;
+        }
+
+        // when the animation completes, adjust the relative layout params of the recycler to make
+        // sure the bar doesn't overlap the bottom row when showing
+        long msDelay = AniUtils.Duration.SHORT.toMillis(this);
+        mHandler.postDelayed(new Runnable() {
+            @Override public void run() {
+                if (!isFinishing()) {
+                    RelativeLayout.LayoutParams params = (RelativeLayout.LayoutParams) mRecycler.getLayoutParams();
+                    if (show) {
+                        params.addRule(RelativeLayout.ABOVE, R.id.container_selection_bar);
+                    } else {
+                        params.addRule(RelativeLayout.ABOVE, 0);
+                    }
+                }
+            }
+        }, msDelay);
+    }
+
+    private void notifySelectionCountChanged() {
+        int numSelected = mAdapter.getSelectionCount();
+        if (numSelected > 0) {
+            String label = String.format(getString(R.string.add_count), numSelected);
+            mTextAdd.setText(label);
+            showSelectionBar(true);
+            if (numSelected == 1) {
+                ActivityUtils.hideKeyboardForced(mSearchView);
+            }
+        } else {
+            showSelectionBar(false);
+        }
+    }
+
+    private void uploadSelection() {
+        ToastUtils.showToast(this, "Uploading will be added in a later PR");
+        // List<StockMediaModel> items = mAdapter.getSelectedStockMedia();
+    }
+
+    class StockMediaAdapter extends RecyclerView.Adapter<StockViewHolder> {
+        private static final float SCALE_NORMAL = 1.0f;
+        private static final float SCALE_SELECTED = .8f;
+
+        private final List<StockMediaModel> mItems = new ArrayList<>();
+        private final ArrayList<Integer> mSelectedItems = new ArrayList<>();
+
+        StockMediaAdapter() {
+            setHasStableIds(true);
+        }
+
+        void setMediaList(@NonNull List<StockMediaModel> mediaList) {
+            mItems.clear();
+            mItems.addAll(mediaList);
+            notifyDataSetChanged();
+        }
+
+        void addMediaList(@NonNull List<StockMediaModel> mediaList) {
+            mItems.addAll(mediaList);
+            notifyDataSetChanged();
+        }
+
+        void clear() {
+            mItems.clear();
+            if (mSelectedItems.size() > 0) {
+                mSelectedItems.clear();
+                notifySelectionCountChanged();
+            }
+            notifyDataSetChanged();
+        }
+
+        @Override
+        public long getItemId(int position) {
+            return mItems.get(position).getId().hashCode();
```

Comment:
Do we need to use the `hashCode`? I'd have guessed the `id` to be enough, but I am guessing it's not unique?

Reply:
Is this situation `id` is a string, so we need to use `hashCode()` to make it a long.

Reply:
Sorry, I am just blind. I even checked the `StockMediaModel` and still didn't see it. 🤦‍♂️